src/main/java/com/aws/greengrass/util/platforms/unix/linux/Cgroup.java

```diff
@@ -63,8 +84,18 @@ public Path getSubsystemComponentPath(String componentName) {
         return getSubsystemGGPath().resolve(componentName);
     }
 
+    /**
+     * get component memory limit path.
+     *
+     * @param componentName componentName
+     * @return memory limit Path
+     */
     public Path getComponentMemoryLimitPath(String componentName) {
-        return getSubsystemComponentPath(componentName).resolve(CGROUP_MEMORY_LIMITS);
+        if (StringUtils.isEmpty(osString)) {
```

Comment:
use `Utils.isEmpty` not StringUtils.

Reply:
What does it mean when the osString is empty? What is the criteria for using MEMORY_MAX instead of CGROUP_MEMORY_LIMITS?

Reply:
When the osString is empty means using the CgroupSubSystemV2,because all the osString in the CgroupSubSystemV2 is empty. We use MEMORY_MAX for v2 and CGROUP_MEMORY_LIMITS for v1.